### ISSUE TITLE
byte, short and negative values

### ISSUE THREAD

**Issue by piluc** [LorenzoBettini/javamm#53]
The following instruction generates a compile error: 
byte x = -1;
It should not, right?
The same happens with the the short data type.

**Comment by LorenzoBettini** [comment]
what is the specific compiler error?
"expected ... but was..." ?

**Comment by LorenzoBettini** [comment]
I confirm that type inference for  this unary expression does not behave as in Java.

Java:

```
byte b = -1;
short s = -1;
short s2 = -10000;
// short s3 = -100000; // Type mismatch: cannot convert from int to short
System.out.println("b = " + b);
System.out.println("s = " + s);
System.out.println("s2 = " + s2);
```

that's because for negative value expression the type expectation is not taken into consideration.

I'll fix that ASAP

**Comment by LorenzoBettini** [comment]
Another Java example

```
static short m() {
	return -1;
}

short s4 = -m(); // Type mismatch: cannot convert from int to short
```

So this bug applies only to values (numeric literals)

**Comment by LorenzoBettini** [comment]
Other Java tests:

```
char c1 = -1; // Type mismatch: cannot convert from int to char
char c2 = +1; // OK
```

**Comment by LorenzoBettini** [comment]
Fixed for some cases, but not for recursive cases: this still fails:

```
	@Test def void testDoubleNegativeCanBeAssignedToShort() {
		// https://github.com/LorenzoBettini/javamm/issues/53
		"short s = -(-10000);".parseAndAssertNoErrors
	}
```

**Comment by LorenzoBettini** [comment]
Now, with the modifications I pushed, it seems to work for all the above cases even for nested unary operations, e.g., -(+1)

However, this will not work

```
short s5 = 1 + -128;
```

and for the moment this specific context is not easily fixable: a huge refactoring of the type system for number literals will be required.

I'll open a new bug to keep track of this type system refactoring.

**Comment by LorenzoBettini** [comment]
I released a new 1.3.4 version with this fix. It should be available in the next few minutes.
